app/models/commit.rb

```diff
@@ -83,6 +83,19 @@ def refresh_statuses!
     end
   end
 
+  def add_status(status_attributes)
+    previous_status = significant_status
+    statuses.create!(status_attributes)
+    reload
+    new_status = significant_status
+    if (previous_status != new_status) # we only care if something changed
```

Comment:
Ah, I understand what you're suggesting now. But only sending the previous status isn't good enough. There could be 3 or 4 or 5 different contexts and the new status may dependent on a historical status. I think we should leave this problem up to the client. Shipit doesn't have enough information to understand the dependencies between contexts.